invoke.go

```diff
@@ -22,16 +22,41 @@ package dig
 
 import (
 	"fmt"
-	"reflect"
-
 	"go.uber.org/dig/internal/digreflect"
 	"go.uber.org/dig/internal/graph"
+	"reflect"
 )
 
-// An InvokeOption modifies the default behavior of Invoke. It's included for
-// future functionality; currently, there are no concrete implementations.
+// An InvokeOption modifies the default behavior of Invoke.
 type InvokeOption interface {
-	unimplemented()
+	applyInvokeOption(*invokeOptions)
+}
+
+type invokeOptions struct {
+	Info *InvokeInfo
+}
+
+// InvokeInfo provides information about an Invoke.
+type InvokeInfo struct {
+	ReqTypes []*Output
```

Comment:
Should we also provide `Input`? (see ProvideInfo)

Reply:
Two things:

- Instead of ReqTypes, is there a fuller name? Inputs or Outputs?
- Are the items inputs or outputs? IIRC, ProvideInfo is Inputs for arguments.

So maybe this should be `Inputs []*Input`?

Reply:
yup, +1 on the naming of the field: let's change this to `Inputs` and `Outputs`.